nano/lib/rep_weights.cpp

```diff
@@ -0,0 +1,69 @@
+#include <nano/lib/rep_weights.hpp>
+#include <nano/secure/blockstore.hpp>
+
+void nano::rep_weights::representation_add (nano::account const & source_rep, nano::uint128_t const & amount_a)
+{
+	std::lock_guard<std::mutex> guard (mutex);
+	auto source_previous (get (source_rep));
+	put (source_rep, source_previous + amount_a);
+}
+
+void nano::rep_weights::representation_put (nano::account const & account_a, nano::uint128_union const & representation_a)
+{
+	std::lock_guard<std::mutex> guard (mutex);
+	return put (account_a, representation_a);
+}
+
+nano::uint128_t nano::rep_weights::representation_get (nano::account const & account_a)
+{
+	std::lock_guard<std::mutex> lk (mutex);
+	return get (account_a);
+}
+
+/** Makes a copy */
+std::unordered_map<nano::account, nano::uint128_t> nano::rep_weights::get_rep_amounts ()
+{
+	std::lock_guard<std::mutex> guard (mutex);
+	return rep_amounts;
+}
+
+void nano::rep_weights::put (nano::account const & account_a, nano::uint128_union const & representation_a)
+{
+	auto it = rep_amounts.find (account_a);
+	auto amount = representation_a.number ();
+	if (it != rep_amounts.end ())
+	{
+		it->second = amount;
+	}
+	else
+	{
+		rep_amounts.emplace (account_a, amount);
+	}
+}
+
+nano::uint128_t nano::rep_weights::get (nano::account const & account_a)
+{
+	auto it = rep_amounts.find (account_a);
+	if (it != rep_amounts.end ())
+	{
+		return it->second;
+	}
+	else
+	{
+		return nano::uint128_t{};
```

Comment:
Is default value 0?

Reply:
Yeh from https://www.boost.org/doc/libs/1_67_0/libs/multiprecision/doc/html/boost_multiprecision/tut/ints/cpp_int.html "Default constructed cpp_int_backends have the value zero." I have added 0 anyway now to remove all doubts :)